### PR TITLE
Basic RabbitMQ queues metricset

### PR DESCRIPTION
**Pull request by kvalev**
Extend the RabbitMQ metricbeat module with a metricset providing basic queue information. The data is retrieved from the RabbitMQ management API.

Related to #3887

### CODE REVIEW

**Comment by exekias** [comment]
jenkins test it

File: metricbeat/module/rabbitmq/queues/_meta/data.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "metricset":{
+        "host":"localhost",
+        "module":"rabbitmq",
+        "name":"queues",
+        "rtt":44269
+    },
+    "rabbitmq":{
+        "queues":{
+            "example": "queues"
+        }
```
Comment (exekias) [inline metricbeat/module/rabbitmq/queues/_meta/data.json]: It looks like some fields are missing from this example `data.json`, can you generate a complete one?

**Comment by kvalev** [comment]
Hello, thanks for the feedback. I have an additional question about the naming conventions - currently the metricset is named 'queues' (as it collects information about queues). This results in the following keys - rabbitmq.queues.name, which looks slightly awkward. Is 'queue' is this case a better name for the metricset?

**Comment by exekias** [comment]
I would say, `queue` looks better for that case

File: metricbeat/module/rabbitmq/queues/_meta/data.json (hunk 1 above)
Comment (kvalev) [inline metricbeat/module/rabbitmq/queues/_meta/data.json]: Done.

File: metricbeat/module/rabbitmq/queue/_meta/fields.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      type: long
+      description: >
+        Number of messages delivered to clients but not yet acknowledged.
+    - name: messages.persistent.count
+      type: long
+      description: >
+        Total number of persistent messages in the queue (will always be 0 for transient queues).
+    - name: memory.bytes
+      type: long
+      description: >
```
Comment (ruflin) [inline metricbeat/module/rabbitmq/queue/_meta/fields.yml]: Could you add here `format: bytes`? Like this Kibana will now it's a bytes value.

Comment (kvalev) [inline metricbeat/module/rabbitmq/queue/_meta/fields.yml]: Done.

**Comment by exekias** [comment]
ok to test

**Comment by exekias** [comment]
LGTM, ready to go, could you add a new entry to `CHANGELOG.asciidoc`?

**Comment by kvalev** [comment]
I have updated the changelog and rebased to the latest master.

**Comment by exekias** [comment]
jenkins, test it please

**Comment by exekias** [comment]
Sorry @kvalev, it took me a while to come back to this, could you please check the errors in https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-intake/1504/console? Basically you need to run `make update` to update docs with the new added fields

**Comment by exekias** [comment]
jenkins retest this please

**Comment by exekias** [comment]
jenkins retest this please

**Comment by ebuildy** [comment]
Just installed it to monitor RabbitMQ 3.3.5, Erlang 17, we dont see any rabbitmq.node.queue.* data:

``` 
"queue": {
                "index": {
                  "read": {},
                  "write": {},
                  "journal_write": {}
                }
              },
              "run": {
                "queue": 0
              },
              "io": {
                "read": {
                  "avg": {}
                },
                "reopen": {},
                "seek": {
                  "avg": {}
                },
                "sync": {
                  "avg": {}
                },
                "write": {
                  "avg": {}
                },
                "file_handle": {
                  "open_attempt": {
                    "avg": {}
                  }
                }
              },
```

How can I debug it? thanks you.

**Pull request merged**